REFERENCE.md

```diff
@@ -138,7 +138,7 @@ module.exports = function(job){
 You can return a value or a promise to signale that the job has been completed.
 
 
-A name argument can be provided so that multiple process functions can be defined per queue. A named process will only process jobs that matches the given name.
+A name argument can be provided so that multiple process functions can be defined per queue. A named process will only process jobs that matches the given name. If you define multiple named process functions in one Queue they will run concurrently (in case jobs are queued for these processors).
 
```

Comment:
hmm, there is a risk for misunderstanding if we say "the will run concurrently" in the sense that they still be in the same queue and processed in a given order (FIFO by default). So for instance if we have a queue like this, with job type A and job type B, the job A will not be processed until all B jobs are processed, but the queue will have a concurrent value of 2:
```AAAAAAABBBBBB```, 

I think that even if this is by design maybe it is not so logical after all...

Reply:
Sorry I didn't understand your comment.

Do you mean that I should further clarify, that even though there may be multiple named process functions in the exact same queue (which may be set to FIFO or LIFO) it doesn't mean it's necessarly going to meet the FIFO/LIFO order? If so then I understood the concept correctly and I can further clarify it.

Reply:
I think that the current behaviour may not be very logical, but anyway it works like this: when you define a process function, you have an optional concurrency parameter, this means that  several processors can run concurrently when processing the queue. But when you define new processors for named jobs, then for every processor you increase the concurrency value, so if you had a concurrency value of 4 in the first processor and a concurrency value of 5 in the second processor now the queue can process 9 jobs concurrently, but it does not matter if they are 9 of the same job name or not.

Reply:
I see, indeed that's not very intuitive. Not even easy to explain :D.

Reply:
By the way would that also mean if someone wants to run 200 concurrent requests in total for all the named processors, he could simply setup ONE process function with a concurrency of 200 and setup the other processor functions with a concurrency of 0? Just for clarification

Reply:
Can you explain why is like that by design? is it posible to change? the whole point of named proccesors is lost - then it behaves like that.

Reply:
@Asp3ctus the behaviour can be changed but why is it so bad as it is now?

Reply:
@manast well as you stated in your comments in this topic you agree this is no very logical.
i will explain my case - of couse their is 100 ways to do anythink - so it's just for example, of cause i can rewrite my logic to that is changed in docs' but personaly i will probably just end up not using bull for my current task - instead make my own much less feture complete implementation.

Let's say you have a queue named "video" - this queue has many task like:
* "record-finder" - find records in db which don't have local processed videos - lets say it selects 10 records then it compleates it creates new job to "video-downloader" - this must run 1 time - only if no other tasks in video "queue"
* "video-downloader" - downloads videos - this can run in 1-4 concurency - so it doesn't use all of the bandwith - then it compleates - it is call "video-converter"
* "video-converter" - this can run in 1-4 concurency (dependant on cpu cores - so the ffmpeg doesn't use all of the system resorces) - then it's done - "db-updater" job is created
* "db-updater" - updates db with converted files - this can have 1-10 concurency - depends on db poll connections
once the db-updater finished - it checks that video queue does not have any other jobs - no mater which named procceeor - is it's empty - first named task is called - if not it will check later
then i finished this case - and started to test it - it ended up eating all of the bandwith and staling my cpu - as it started more and more ffmpeg proccesses as it progressed -it took me a lot of time to find this "feature/bug" before you updated the docs - and i was very disapointed then it ended up beeing a feature - not a bug - as it does not fit in my religion :) 

and this is only explain one queue - i will have 5-6 of diferent types  - this is very neat to organaize it in queus and create named proccesors - which have it's on concurency setting

if i will go the way a new documentation is stated - i will end up having 20-30 queues instead of 5-6 - the code will be more complex - the administration will be more coplex - more connections to redis(but it's not the issue)

My point is that if each named proccessor will follow it's own concurency settings - it will give more power to bull - and probably make some other person happy - not just me :)